scio-test/src/test/scala/com/spotify/scio/testing/JobTestTest.scala

```diff
@@ -296,6 +310,26 @@ class JobTestTest extends PipelineSpec {
     an [AssertionError] should be thrownBy { testDistCacheJob("a1", "a2", "b1", "b2", "c3", "d4") }
   }
 
+  def testCustomIOJob(xs: Int*): Unit = {
+    JobTest[CustomIOJob.type]
+      .args("--input=in.txt", "--output=out.txt")
+      .input(CustomIO("TextIn"), Seq(1, 2, 3))
```

Comment:
You probably could move `inputTransform` to field inside `CustomIOJob` (with `named("TextIn ")`) and then here just get name without - thus reduce the duplicate of names etc. The same for `outputTransform`.